subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/DefaultAdhocSoftwareComponent.java

```diff
@@ -73,6 +73,10 @@ public Set<? extends UsageContext> getUsages() {
         return builder.build();
     }
 
+    protected boolean isPublished(Configuration outgoingConfiguration) {
```

Comment:
❕ I think you should remove this and expose the `variants` collection as protected or provide a getter for it, and do the test directly in the `withX` methods.  The name is misleading - it may or may not have published the variant based on this configuration, and it's not the conf being published anyway - it's the variant.